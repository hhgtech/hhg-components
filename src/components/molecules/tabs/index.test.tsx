import React from 'react'

import { render, cleanup } from '@testing-library/react'
import 'jest-styled-components'

import { Tabs } from '.'
import MockTheme from './../../../mockTheme'

afterEach(cleanup)

test('it renders', () => {
  const tabs = render(
    <MockTheme>
      <Tabs
        initialTab={0}
        tabContent={[
          {
            name: 'Tab 1',
          },
          {
            name: 'With icon',
            icon: <img src="//unsplash.it/16/16" />,
          },
          {
            name: 'Tab 2',
          },
          {
            name: 'With icon',
            icon: <img src="//unsplash.it/16/16" />,
          },
          {
            name: 'With icon',
            icon: <img src="//unsplash.it/16/16" />,
          },
          {
            name: 'With icon',
            icon: <img src="//unsplash.it/16/16" />,
          },
        ]}
      />
    </MockTheme>,
  )

  const self = tabs.container.firstChild.firstChild

  expect(self).toMatchSnapshot()

  expect(self).toHaveAttribute('data-is-tab-active', 'true')
})
