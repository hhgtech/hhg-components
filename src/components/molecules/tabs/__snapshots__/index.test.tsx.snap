// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders 1`] = `
.c2 {
  --default-color: #262626;
  --text-color: var(--custom-color,--default-color);
  color: var(--text-color);
}

.c2[data-type='bold'],
.c2[data-variant='heading'],
.c2[data-type='caption'] {
  font-weight: 700;
}

.c2[data-type='caption'] {
  text-transform: uppercase;
}

.c0 {
  padding-bottom: 4px;
}

.c0 p {
  color: #8c8c8c;
  white-space: nowrap;
}

.c0:not(:last-child) {
  margin-right: 24px;
}

.c0[data-is-tab-active] {
  border-bottom: 2px solid #2d87f3;
  margin-bottom: -1px;
}

.c0[data-is-tab-active] p {
  color: #2d87f3;
}

.c0:not([data-is-tab-active]) {
  cursor: pointer;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

@media (min-width:768px) {
  .c2[data-size='lg'] {
    font-size: 26px;
    -webkit-letter-spacing: -0.4px;
    -moz-letter-spacing: -0.4px;
    -ms-letter-spacing: -0.4px;
    letter-spacing: -0.4px;
    line-height: 1.2;
  }

  .c2[data-size='md'] {
    font-size: 20px;
    -webkit-letter-spacing: -0.4px;
    -moz-letter-spacing: -0.4px;
    -ms-letter-spacing: -0.4px;
    letter-spacing: -0.4px;
    line-height: 1.2;
  }

  .c2[data-size='base'] {
    font-size: 16px;
    line-height: 1.5;
  }

  .c2[data-size='sm'] {
    font-size: 13px;
    line-height: 1.5;
  }

  .c2[data-size='xs'] {
    font-size: 11px;
    line-height: 1.5;
  }
}

@media (max-width:767px) {
  .c2[data-size='lg'] {
    font-size: 22px;
    -webkit-letter-spacing: -0.4px;
    -moz-letter-spacing: -0.4px;
    -ms-letter-spacing: -0.4px;
    letter-spacing: -0.4px;
    line-height: 1.2;
  }

  .c2[data-size='md'] {
    font-size: 16px;
    line-height: 1.2;
  }

  .c2[data-size='base'] {
    font-size: 14px;
    line-height: 1.5;
  }

  .c2[data-size='sm'] {
    font-size: 13px;
    line-height: 1.5;
  }

  .c2[data-size='xs'] {
    font-size: 11px;
    line-height: 1.5;
  }
}

<li
  class="c0"
  data-is-tab-active="true"
>
  <div
    class="c1"
  >
    <p
      class="c2"
      data-size="sm"
      data-type="bold"
    >
      Tab 1
    </p>
  </div>
</li>
`;
